File: jambeth/jAmbeth.Test/test/de/osthus/ambeth/relations/one/lazy/fk/reverse/none/EntityB.java
package de.osthus.ambeth.relations.one.lazy.fk.reverse.none;

import de.osthus.ambeth.model.AbstractEntity;

public class EntityB extends AbstractEntity
{
	protected String name;

	protected EntityA entityA;

	protected EntityB()
	{
		// Intended blank
	}

	public String getName()
	{
		return name;
	}

	public void setName(String name)
	{
		this.name = name;
	}

	public EntityA getEntityA()
	{
		return entityA;
	}

	public void setEntityA(EntityA entityA)
	{
		this.entityA = entityA;
	}
}

File: ambeth/Ambeth.Test/ambeth/bytecode/EntityB.cs
using De.Osthus.Ambeth.Merge;
using System;

namespace De.Osthus.Ambeth.Bytecode
{
    /**
     * Entity based on a class with default and non default constructor
     */
    public abstract class EntityB : EntityA
    {
        protected EntityB()
        {
            // Intended blank
        }

        protected EntityB(String name, String value)
        {
            this.name = name;
            this.value = value;
        }
    }
}